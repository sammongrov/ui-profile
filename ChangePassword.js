import React, { Component } from 'react';
import { View, TouchableOpacity, Alert, BackHandler } from 'react-native';
import { styles } from 'react-native-theme';
import { Actions } from 'react-native-router-flux';
import { Colors } from '@ui/theme_default';
import { NavBar, Screen, Icon, TextInput, Button, Text } from '@ui/components';
// import DBManager from '../app/DBManager';
import {DBManager} from 'app-module';

export default class ChangePassword extends Component {
  state = {
    newPassword: '',
    confirmPassword: '',
  };

  componentWillMount() {
    BackHandler.addEventListener('hardwareBackPress', this.handleBackPress);
  }

  componentWillUnmount = () => {
    BackHandler.removeEventListener('hardwareBackPress', this.handleBackPress);
  };

  handleBackPress = () => {
    Actions.pop();
    return true;
  };

  resetCallBack = (result, reason) => {
    if (result) {
      DBManager.app.logout();
      Actions.Login();
      // Alert.alert("Password changed")
    } else {
      Alert.alert(reason);
    }
  };

  updatePassword = (newPassword, confirmPassword) => {
    if (newPassword !== confirmPassword) {
      Alert.alert('Password not match');
    } else {
      DBManager.app.resetPassword(newPassword, this.resetCallBack);
    }
  };

  render() {
    const { newPassword, confirmPassword } = this.state;
    return (
      <Screen>
        <NavBar
          titleText="Change Password"
          leftComponent={
            <TouchableOpacity
              onPress={Actions.pop}
              style={[styles.navSideButtonDimension, styles.alignJustifyCenter]}
            >
              <Icon
                name="chevron-left"
                type="material-community"
                color={Colors.NAV_ICON}
                size={36}
              />
            </TouchableOpacity>
          }
        />
        <View style={styles.changePasswordContainer}>
          <View style={styles.columnDirection}>
            <Text style={[styles.authTFLabelView, styles.changePasswordTitleStyle]}>
              New Password
            </Text>
            <TextInput
              onChangeText={(text) => this.setState({ newPassword: text })}
              autoCapitalize="none"
              style={styles.authTFInput}
              value={newPassword}
              underlineColorAndroid="transparent"
              autoCorrect={false}
              clearButtonMode="while-editing"
              secureTextEntry
              disableFullscreenUI
            />
          </View>
          <View style={[styles.columnDirection, styles.marginTop20]}>
            <Text style={[styles.authTFLabelView, styles.changePasswordTitleStyle]}>
              Confirm Password
            </Text>
            <TextInput
              onChangeText={(text) => this.setState({ confirmPassword: text })}
              autoCapitalize="none"
              style={[styles.authTFInput]}
              value={confirmPassword}
              underlineColorAndroid="transparent"
              autoCorrect={false}
              clearButtonMode="while-editing"
              secureTextEntry
              disableFullscreenUI
            />
          </View>
          <Button
            title="Update Password"
            onPress={() => this.updatePassword(newPassword, confirmPassword)}
            color={Colors.BG_BTN}
            buttonStyle={[styles.serverConfirmButton]}
            containerStyle={[styles.marginTop15]}
          />
        </View>
      </Screen>
    );
  }
}
