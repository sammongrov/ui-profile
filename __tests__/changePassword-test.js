import React from 'react';
import { Alert, BackHandler } from 'react-native';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Actions } from 'react-native-router-flux';
import { iOSColors } from 'react-native-typography';
import ChangePassword from '../ChangePassword';
import DbManager from '../../app/DBManager';

configure({ adapter: new Adapter() });
const onConfirmServer = jest.fn();
DbManager.app.logout = jest.fn();
DbManager.app.resetPassword = jest.fn();

jest.mock('react-native-router-flux', () => ({
  Actions: {
    pop: jest.fn(),
    Login: jest.fn(),
  },
}));

jest.mock('BackHandler', () => {
  const backHandler = {
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
  };
  return backHandler;
});

jest.mock('Alert', () => {
  const alert = {
    alert: jest.fn(),
  };
  return alert;
});

jest.mock('../../app/DBManager', () => {
  const dbManager = {
    app: {},
    user: {},
  };
  return dbManager;
});

it('render correctly with props', () => {
  const tree = renderer.create(<ChangePassword onConfirmServer={onConfirmServer} />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('should render the onChange to input newPassword', () => {
  const rootComponent = shallow(<ChangePassword />);
  const authorInputComponent = rootComponent.find('TextInput').first();

  authorInputComponent.simulate('ChangeText', 'text');
  expect(rootComponent.state('newPassword')).toEqual('text');
});

it('should render the onChange to input confirmPassword', () => {
  const rootComponent = shallow(<ChangePassword />);
  const authorInputComponent = rootComponent.find('TextInput').last();

  authorInputComponent.simulate('ChangeText', 'text');
  expect(rootComponent.state('confirmPassword')).toEqual('text');
});

it('confirm button is clicked', () => {
  const rootComponent = shallow(<ChangePassword />);
  const instance = rootComponent.instance();
  instance.updatePassword = jest.fn();

  const button = rootComponent.find('Button').first();
  button.props().onPress();
  expect(instance.updatePassword.mock.calls.length).toEqual(1);
});

it('ChangePassword unmounts correctly', () => {
  const tree = shallow(<ChangePassword />);
  tree.unmount();
  expect(BackHandler.addEventListener).toHaveBeenCalled();
});

it('resetCallBack', () => {
  const rootComponent = shallow(<ChangePassword />);
  const instance = rootComponent.instance();
  const result = true;
  const reason = 'test reason';

  instance.resetCallBack(result, reason);
  expect(DbManager.app.logout.mock.calls.length).toEqual(1);
  expect(Actions.Login.mock.calls.length).toEqual(1);
});

it('else resetCallBack', () => {
  const rootComponent = shallow(<ChangePassword />);
  const instance = rootComponent.instance();
  const result = false;
  const reason = false;
  instance.resetCallBack(result, reason);
  expect(Alert.alert).toHaveBeenCalledWith(reason);
});

it('updatePassword', () => {
  const rootComponent = shallow(<ChangePassword />);
  const instance = rootComponent.instance();
  const newPassword = true;
  const confirmPassword = 'test confirmPassword';

  instance.updatePassword(newPassword, confirmPassword);
  expect(Actions.Login.mock.calls.length).toEqual(1);
});

it('else updatePassword', () => {
  const rootComponent = shallow(<ChangePassword />);
  const instance = rootComponent.instance();
  const newPassword = true;
  const confirmPassword = true;

  instance.updatePassword(newPassword, confirmPassword);
  expect(DbManager.app.resetPassword.mock.calls.length).toEqual(1);
});

it('handleBackPress', () => {
  const rootComponent = shallow(<ChangePassword />);
  const instance = rootComponent.instance();
  const result = instance.handleBackPress();

  expect(result).toBe(true);
  expect(Actions.Login.mock.calls.length).toBe(1);
});
