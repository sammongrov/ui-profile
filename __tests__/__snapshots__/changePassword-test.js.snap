// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render correctly with props 1`] = `
<BVLinearGradient
  colors={
    Array [
      4293914607,
      4293914607,
    ]
  }
  locations={
    Array [
      0,
      1,
    ]
  }
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RCTSafeAreaView
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      onLayout={[Function]}
    >
      <View
        style={
          Array [
            undefined,
            null,
          ]
        }
      >
        <View
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <Text
              accessible={true}
              allowFontScaling={false}
              ellipsizeMode="tail"
              style={
                Array [
                  Object {
                    "color": "#2E88FF",
                    "fontSize": 36,
                  },
                  undefined,
                  Object {
                    "fontFamily": "Material Design Icons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
              type="material-community"
            >
              ÔÖÅ
            </Text>
          </View>
        </View>
        <View
          style={
            Array [
              undefined,
              undefined,
              null,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            numberOfLines={1}
            style={
              Array [
                Object {
                  "fontFamily": "OpenSans-Regular",
                },
                Array [
                  undefined,
                  null,
                ],
              ]
            }
          >
            Change Password
          </Text>
        </View>
        <View
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        />
      </View>
      <View>
        <View>
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": "OpenSans-Regular",
                },
                Array [
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            New Password
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            autoCorrect={false}
            clearButtonMode="while-editing"
            disableFullscreenUI={true}
            onChangeText={[Function]}
            secureTextEntry={true}
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <View
          style={
            Array [
              undefined,
              undefined,
            ]
          }
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Array [
                Object {
                  "fontFamily": "OpenSans-Regular",
                },
                Array [
                  undefined,
                  undefined,
                ],
              ]
            }
          >
            Confirm Password
          </Text>
          <TextInput
            allowFontScaling={true}
            autoCapitalize="none"
            autoCorrect={false}
            clearButtonMode="while-editing"
            disableFullscreenUI={true}
            onChangeText={[Function]}
            secureTextEntry={true}
            style={
              Array [
                undefined,
              ]
            }
            underlineColorAndroid="transparent"
            value=""
          />
        </View>
        <View
          style={
            Array [
              Array [
                undefined,
              ],
              false,
            ]
          }
        >
          <View
            accessible={true}
            isTVSelectable={true}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "opacity": 1,
              }
            }
          >
            <View
              style={
                Array [
                  Object {
                    "alignItems": "center",
                    "backgroundColor": "#2089dc",
                    "borderRadius": 3,
                    "flexDirection": "row",
                    "justifyContent": "center",
                  },
                  Array [
                    undefined,
                  ],
                  false,
                  false,
                  false,
                  undefined,
                ]
              }
            >
              <Text
                accessible={true}
                allowFontScaling={true}
                ellipsizeMode="tail"
                style={
                  Array [
                    Object {
                      "backgroundColor": "transparent",
                      "color": "white",
                      "fontSize": 18,
                      "padding": 8,
                      "textAlign": "center",
                    },
                    undefined,
                    false,
                    false,
                  ]
                }
              >
                Update Password
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RCTSafeAreaView>
</BVLinearGradient>
`;
